### PR TITLE
Implement update level crossing

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I don't think this function should be under ObjectManager but not sure where it should go. Also the name isn't great.

### CODE REVIEW

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 1]
```diff
@@ -936,8 +938,45 @@ namespace OpenLoco::ObjectManager
     // 0x004796A9
     void updateYearly1()
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:939]: Can we rename this to `updateLevelCrossingDesign`?

Comment (LeftofZen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:939]: playtesting note - level crossings already places on the map do not update past their 'designed year' like streetlights - this code only determines what crossing type is used for *new* crossings.

Comment (LeftofZen) [reply src/OpenLoco/src/Objects/ObjectManager.cpp:939]: `defaultLevelCrossing`?

File: src/OpenLoco/src/GameState.h [hunk 2]
```diff
@@ -63,7 +63,7 @@ namespace OpenLoco
         uint32_t orderTableLength;                                               // 0x0001A0 (0x00525FB8)
         uint32_t roadObjectIdIsTram;                                             // 0x0001A4 (0x00525FBC)
         uint32_t roadObjectIdIsFlag7;                                            // 0x0001A8 (0x00525FC0)
-        uint8_t var_1AC;                                                         // 0x0001AC (0x00525FC4)
+        uint8_t currentLevelCrossingType;                                        // 0x0001AC (0x00525FC4)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameState.h:66]:
```suggestion
        uint8_t currentDefaultLevelCrossingType;                                 // 0x0001AC (0x00525FC4)
```

File: src/OpenLoco/src/Objects/ObjectManager.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/Objects/ObjectManager.cpp:939]:
```suggestion
    void updateDefaultLevelCrossingType()
```

File: src/OpenLoco/src/Objects/ObjectManager.cpp [hunk 3]
```diff
@@ -930,10 +932,54 @@ namespace OpenLoco::ObjectManager
     }
 
     // 0x004796A9
-    void updateYearly1()
+    void updateDefaultLevelCrossingType()
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Objects/ObjectManager.cpp:935]: You've already indicated the ObjectManager isn't a great place for this. Perhaps the Scenario namespace would be a better fit?

**Review by AaronVanGeffen — COMMENTED** [review]
Code looks fine, and the transition seems to work properly. (Left: post-1940, right: pre-1940)

Screenshot 2024-03-22 at 20 40 24

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**